sensing subsystem update

ghu0510 opened the pull request:
sensing: refine sensing API
sensing: add hinge angle sensor skeleton
sensing: implement set/get interval and sensitivity
sensing: create runtime thread and implement later config

ghu0510 committed:
sensing: add hinge angle sensor skeleton

Add hinge angle virtual sensor for sensing subsystem, hinge angle sensor
takes both base accel and lid accel as reporter

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

ghu0510 committed:
sensing: implement set/get interval and sensitivity

set/get interval sensitivity is implmented in sensor management module,
and also add related test case

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

ghu0510 committed:
sensing: create runtime thread and implement later config

create runtime thread and wait event semphore, meanwhile, trigger event
semphore in set interval/sensitivity API to notify runtime thread to
arbitrate interval/sensitivity

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

yperess commented:
During the review of https://github.com/zephyrproject-rtos/zephyr/pull/55389 it was agreed to unblock it as long as design concerns are addressed before any additional changes are built on top of the subsystem. Please address https://github.com/zephyrproject-rtos/zephyr/issues/58925 first.

ghu0510 committed:
sensing: refine sensing API

1) modify reporter type from int to const sensing_sensor_handle_t in
sensing_sensor_process_t callback
2) move variable shift behind struct sensing_sensor_value_header in
struct sensing_sensor_value_q31
3) add pointer checking for sensing API

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>
